packages/cloud_firestore/lib/src/blob.dart

```diff
@@ -0,0 +1,27 @@
+// Copyright 2018, the Chromium project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+part of cloud_firestore;
+
+class Blob {
+  final Uint8List bytes;
+  const Blob(this.bytes);
+
+  @override
+  bool operator ==(dynamic other) =>
+      other is Blob && other.hashCode == hashCode;
+
+  @override
+  int get hashCode {
+    return hashCodeIteration(0, bytes);
+  }
+
+  int hashCodeIteration(int hashBase, Iterable<int> byteList) {
+    if (byteList.isNotEmpty) {
+      return hashCodeIteration(
+          hashValues(hashBase, byteList.first), byteList.skip(1));
```

Comment:
Doing a recursive call and constructing a new `Iterable` per byte is unlikely to perform well (AFAIK, Dart does not implement tail call optimization, https://github.com/dart-lang/sdk/issues/29).

You should be able to do
```dart
int get hashCode => hashList(bytes);
```
with `hashList` imported from `dart:ui`.